src/emulator/gui/src/ui_allocations_dialog.cpp

```diff
@@ -0,0 +1,61 @@
+#include <gui/functions.h>
+#include <gui/gui_constants.h>
+
+#include <imgui.h>
+#include <imgui_memory_editor.h>
+
+#include <host/state.h>
+
+#include <string.h>
+
+void DrawAllocationsDialog(HostState &host) {
+    ImGui::Begin("Allocations", &host.gui.allocations_dialog);
+
+    const std::lock_guard<std::mutex> lock(host.mem.generation_mutex);
+    for (const auto &pair : host.mem.generation_names) {
+        int length = snprintf(nullptr, 0, "%i: %s", (int)pair.first, pair.second.c_str());
```

Comment:
```suggestion
        int length = snprintf(nullptr, 0, "%zu: %s", pair.first, pair.second.c_str());
```
For string formatting you could've also used [`fmt::format()`](http://fmtlib.net/latest/syntax.html), with the format string simply being `"{}: {}"`.